### PR TITLE
feat(text-input-react): add autoExpand to TextArea

### PR DESCRIPTION
**Pull request by Saegrov**
affects: @fremtind/jkl-text-input-react

Add the ability for TextArea to increase in height as you write.

ISSUES CLOSED: #769

## 📥 Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## ☑️ Submission checklist

-   [x] I have read the [CONTRIBUTING](https://github.com/fremtind/jokul/blob/master/CONTRIBUTING.md) doc
-   [x] `yarn build` works locally with my changes
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] I have added necessary documentation (if appropriate)

## 💬 Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did, what alternatives you considered, etc...

### CODE REVIEW

File: packages/text-input-react/src/TextArea.tsx [hunk 1]
```diff
@@ -20,26 +21,79 @@ export const TextArea = ({
...
             </Label>
             <textarea
+                onFocus={(e) => {
+                    setTextAreaFocused(true);
+                    restProps.onFocus && restProps.onFocus(e);
+                }}
+                onBlur={(e) => {
+                    setTextAreaFocused(false);
+                    restProps.onBlur && restProps.onBlur(e);
+                }}
```
Comment (piofinn) [inline packages/text-input-react/src/TextArea.tsx]: Trekke ut til navngitte funksjoner, kanskje?

File: packages/text-input-react/src/TextArea.tsx [hunk 2]
```diff
@@ -20,26 +21,79 @@ export const TextArea = ({
...
     forceCompact,
     ...restProps
 }: Props) => {
     const componentClassName = classNames("jkl-text-field jkl-text-area", className, {
         "jkl-text-field--compact": forceCompact,
     });
 
+    const [textAreaFocused, setTextAreaFocused] = useState(false);
+    const textAreaRef = useRef<HTMLTextAreaElement>(null);
+    const textAreaContentRows = (restProps.value || "").split("\n").length;
```
[the remark this answers is not in the document]
Comment (Saegrov) [reply packages/text-input-react/src/TextArea.tsx]: Ikke i nyere browsere i følge denne:
https://stackoverflow.com/a/6833783

File: packages/text-input-react/src/TextArea.tsx (hunk 1 above)
Comment (Saegrov) [reply packages/text-input-react/src/TextArea.tsx]: Aye.

File: packages/text-input-react/src/TextArea.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (vegardsoerlie) [reply packages/text-input-react/src/TextArea.tsx]: > Er kanskje ingen plattformer som kun bruker \r som linjeskift?

Apple II? 😅

**Pull request merged**